### ISSUE TITLE
Too easy to accidentally abort a rebase by hitting esc when there are conflicts

### ISSUE THREAD

**Issue by stefanhaller** [jesseduffield/lazygit#2678]
When a rebase stops with conflicts, there's the prompt `Conflicts! To abort press 'esc', otherwise press 'enter'`. It's too easy to accidentally hit escape here and abort a rebase, which can be super annoying when you already spent a long time resolving conflicts in earlier commits.

It's especially a problem because at the prompt `all merge conflicts resolved. Continue?` I usually do hit escape.

I don't have a great idea what to do about it; but a simple, additional `Are you sure you want to abort the rebase?` prompt would do, I guess.

**Comment by stefanhaller** [comment]
A solution could be as simple as this:

```diff
diff --git a/pkg/gui/controllers/helpers/merge_and_rebase_helper.go b/pkg/gui/controllers/helpers/merge_and_rebase_helper.go
index 0e67fd7ea..4a7fd0442 100644
--- a/pkg/gui/controllers/helpers/merge_and_rebase_helper.go
+++ b/pkg/gui/controllers/helpers/merge_and_rebase_helper.go
@@ -145,7 +145,7 @@ func (self *MergeAndRebaseHelper) CheckMergeOrRebase(result error) error {
 				return self.c.PushContext(self.c.Contexts().Files)
 			},
 			HandleClose: func() error {
-				return self.genericMergeCommand(REBASE_OPTION_ABORT)
+				return self.AbortMergeOrRebaseWithConfirm()
 			},
 		})
 	} else {
```

Any opinions @jesseduffield?

**Comment by jesseduffield** [comment]
I agree that the current UX is bad. Looking for `HandleClose` in the codebase this really is the only place where we perform an action upon hitting escape in a popup.

As a rule, if 'I don't care' is a permitted response to a popup, the user should be able to hit escape and no action should be performed (with focus returning to the original context)

We should have a menu with the following two options (and no cancel button because it's not really appropriate)
* v - View conflicts
* a - Abort

Choosing 'View conflicts' will focus the files context, choosing 'Abort' will abort, and hitting escape will return focus to the original context.

This UX is backwards compatible in terms of viewing the conflicts (you're hitting enter either way) but not in terms of aborting (because escape now does something different). That's not a big deal though: if somebody habitually hits escape they can still abort by pressing 'm'.

Also, with this new UX you'll also be able to abort with a single keypress.

What do you think?

**Comment by stefanhaller** [comment]
I like the suggestion, I'll see if I can find the time to work on a PR.

**Comment by stefanhaller** [comment]
PR is here: #2694
